### ISSUE TITLE
Docker-compose "ERROR: Encountered errors while bringing up the project."

### ISSUE THREAD

**Issue by Gufderald** [grafana/oncall#239]
### Discussed in https://github.com/grafana/oncall/discussions/237

<div type='discussions-op-text'>

<sup>Originally posted by **Gufderald** July 15, 2022</sup>
Hi!
While trying to start hobby playground environment, I've faced with next error on `docker-compose --env-file .env_hobby -f docker-compose.yml up --build -d` stage:
```
Creating network "oncall_default" with the default driver
Creating volume "oncall_dbdata" with default driver
Creating volume "oncall_rabbitmqdata" with default driver
Creating oncall_mysql_1    ... done
Creating oncall_redis_1    ... done
Creating oncall_rabbitmq_1 ... done
Creating oncall_oncall_db_migration_1        ... done
Creating oncall_mysql_to_create_grafana_db_1 ... done
Creating oncall_grafana_1                    ... done

ERROR: for engine  Container "5da314125e65" exited with code 1.

ERROR: for celery  Container "5da314125e65" exited with code 1.
ERROR: Encountered errors while bringing up the project.
```
I can also see `kombu.exceptions.OperationalError: [Errno 111] Connection refused` in `oncall_db_migration` logs.

Howewer, after manual restart of `oncall_db_migration` and one more `docker-compose --env-file .env_hobby -f docker-compose.yml up --build -d`, everything works fine. Am I missing something?
</div>

**Comment by Matvey-Kuk** [comment]
@Gufderald thank you for reporting this! Sounds like `oncall_db_migration ` starts before rabbit. We have healthcheck for MySQL, let me see if I can make smth like this for rabbit.

**Comment by Matvey-Kuk** [comment]
@Gufderald should be fixed, thank you for reporting!

**Comment by Gufderald** [comment]
Should I open another issue for that?

**Comment by Gufderald** [comment]
Actually yes. Previously I've tested it both on our main node (64 threads/32GB ram) and VM(2 threads, 4GB ram), and they had same behaviour. Now node is buisy, so I tested only on VM. I'll be able to try it later, and report it here. Thank you for your reply!

**Comment by Gufderald** [comment]
@Matvey-Kuk Hi again! Tested it on a powerful machine with updated `docker-compose.yml`, and got almost same behaviour as at the beginning of this tread: 
```
Creating oncall_redis_1    ... done
Creating oncall_rabbitmq_1 ... done
Creating oncall_mysql_1    ... done
Creating oncall_mysql_to_create_grafana_db_1 ... done
Creating oncall_grafana_1                    ... done
Creating oncall_oncall_db_migration_1        ... done

ERROR: for celery  Container "8ba983e27132" exited with code 1.

ERROR: for engine  Container "8ba983e27132" exited with code 1.
ERROR: Encountered errors while bringing up the project.
```
I thought I shouldn't open another issue for this, as behaviour seems to be related to this issue.
However, now `celery` and `engine` errors follows in opposite order, if that means something.
I must also mention, that this is machine with our previous iteration of oncall, and I made only `docker-compose --env-file .env_hobby -f docker-compose.yml down` and `docker-compose --env-file .env_hobby -f docker-compose.yml pull`, without removing volumes. Can this be a reason? Thank you in advance for your time.

**Comment by Matvey-Kuk** [comment]
@Gufderald oh wow! Do you have logs from those failing containers? It will be really helpful.

**Comment by Matvey-Kuk** [comment]
Closing.

**Comment by Matvey-Kuk** [comment]
Reproduced, reopening.

**Comment by th30nlyw4y** [comment]
Looks like https://github.com/grafana/oncall/pull/240 didn't help (at least for me), RabbitMQ takes about ~1 minute to start, so i've changed config to :
```
    healthcheck:
      test: rabbitmq-diagnostics -q ping
      start_period: 1m
      interval: 30s
      timeout: 10s
      retries: 3
```
As for the existing problem, i see this logs in `oncall_db_migration` container:
```
amqp.exceptions.AccessRefused: (0, 0): (403) ACCESS_REFUSED - Login was refused using authentication mechanism PLAIN. For details see the broker logfile
```
And in RabbitMQ container:
```
2022-09-08 21:07:15.866 [info] <0.578.0> accepting AMQP connection <0.578.0> (172.24.0.5:34096 -> 172.24.0.4:5672)
2022-09-08 21:07:15.867 [error] <0.578.0> Error on AMQP connection <0.578.0> (172.24.0.5:34096 -> 172.24.0.4:5672, state: starting):
PLAIN login refused: user 'rabbitmq' - invalid credentials
2022-09-08 21:07:15.867 [info] <0.578.0> closing AMQP connection <0.578.0> (172.24.0.5:34096 -> 172.24.0.4:5672)
```
But credentials are the same. RabbitMQ:
```


```
Migration container's env:
```
BASE_URL=host.docker.internal
SECRET_KEY=6HupX9QT7tGQSinC6ccnH1VaUH8TgRQf
RABBITMQ_USERNAME=rabbitmq
RABBITMQ_PASSWORD=1zYOEEgN9QOO7b11EZf5sZ1aOdBiQ8L4
RABBITMQ_HOST=rabbitmq
RABBITMQ_PORT=5672
RABBITMQ_DEFAULT_VHOST=/
MYSQL_PASSWORD=12345
MYSQL_DB_NAME=oncall_hobby
MYSQL_USER=root
MYSQL_HOST=mysql
MYSQL_PORT=3306
REDIS_URI=redis://redis:6379/0
DJANGO_SETTINGS_MODULE=settings.hobby
OSS=True
CELERY_WORKER_QUEUE=default,critical,long,slack,telegram,webhook,retry,celery
```
Maybe i'm missing some configuration details ?

**Comment by scubbo** [comment]
I'm getting this same intermittent issue (Macbook Pro, Mac OS 12.6, Intel Core i7; Docker compose version 2.7.0). Let me know if I can do anything to help with diagnosis.

**Comment by scubbo** [comment]
I can typically resolve this with the following command defined in Makefile:

```
super-restart:
  make start; sleep 10 && docker restart oncall_db_migration && sleep 10 && docker restart oncall_engine oncall_celery grafana oncall_ui
```

(Note that the indent _must_ be a tab character and not spaces!)

**Comment by joeyorlando** [comment]
@Gufderald & @th30nlyw4y 
#802 _may_ have fixed your issue. Basically it bumped the `docker-compose` file version to `3.9` (from `3.8`) which supports the long-hand form of `depends_on` (see [here](https://stackoverflow.com/a/54249757) for more context).

Let me know if you're still experiencing this.

**Comment by joeyorlando** [comment]
also, see [this comment](https://github.com/grafana/oncall/issues/846#issuecomment-1323322508) in #846.

tldr; you may need to upgrade `docker-compose` to the latest version. `docker-compose` v.2.10 patched some issues related to the usage of `depends_on`.

**Comment by scubbo** [comment]
Success! After updating my docker-compose version from 2.7.0, this works:

```
$ git clone git@github.com:grafana/oncall.git oncall-fresh-clone
Cloning into 'oncall-fresh-clone'...
[...]
$ docker compose version
Docker Compose version v2.12.2
$ cd oncall-fresh-clone
~/Code/oncall-fresh-clone
$ make init start
[...success...]
```

Thanks for your help!

**Comment by joeyorlando** [comment]
closing as this should be addressed with the above recommendation (has been documented in #885). If anyone is still running into this, feel free to open another issue.
